tests/testthat/test_matrix_normalization.R

```diff
@@ -0,0 +1,43 @@
+test_that(
+  "Test matrix normalization stan function and its purpose.",
+  {
+    model <- compiled_site_inf_model
+    space_model_fxns <- spatial_fxns$functions
+
+
+    sigma2_generalized <- 0.2
+    dist_matrix <- as.matrix(
+      dist(
```

Comment:
I think a good test here would be to modify the scale of the distance matrix (e.g. multiply by a scale factor) and then ensure that both normalized matrices are equivalent! Since I think this is the use case we want this for if I am understanding correctly! Would just suggest adding that additional test here.